erpnext/accounts/doctype/payment_request/test_payment_request.py

```diff
@@ -415,3 +417,256 @@ def test_conversion_on_foreign_currency_accounts(self):
 		self.assertEqual(pe.paid_amount, 800)
 		self.assertEqual(pe.base_received_amount, 800)
 		self.assertEqual(pe.received_amount, 10)
+
+	def test_multiple_payment_if_partially_paid_for_same_currency(self):
+		so = make_sales_order(currency="INR", qty=1, rate=1000)
+
+		self.assertEqual(so.advance_payment_status, "Not Requested")
+
+		pr = make_payment_request(
+			dt="Sales Order",
+			dn=so.name,
+			mute_email=1,
+			submit_doc=1,
+			return_doc=1,
+		)
+
+		self.assertEqual(pr.grand_total, 1000)
+		self.assertEqual(pr.outstanding_amount, pr.grand_total)
+		self.assertEqual(pr.party_account_currency, pr.currency)  # INR
+		self.assertEqual(pr.status, "Requested")
+
+		so.load_from_db()
+		self.assertEqual(so.advance_payment_status, "Requested")
+
+		# to make partial payment
+		pe = pr.create_payment_entry(submit=False)
+		pe.paid_amount = 200
+		pe.references[0].allocated_amount = 200
+		pe.submit()
+
+		self.assertEqual(pe.references[0].payment_request, pr.name)
+
+		so.load_from_db()
+		self.assertEqual(so.advance_payment_status, "Partially Paid")
+
+		pr.load_from_db()
+		self.assertEqual(pr.status, "Partially Paid")
+		self.assertEqual(pr.outstanding_amount, 800)
+		self.assertEqual(pr.grand_total, 1000)
+
+		# complete payment
+		pe = pr.create_payment_entry()
+
+		self.assertEqual(pe.paid_amount, 800)  # paid amount set from pr's outstanding amount
+		self.assertEqual(pe.references[0].allocated_amount, 800)
+		self.assertEqual(pe.references[0].outstanding_amount, 800)  # for Orders it is not zero
+		self.assertEqual(pe.references[0].payment_request, pr.name)
+
+		so.load_from_db()
+		self.assertEqual(so.advance_payment_status, "Fully Paid")
+
+		pr.load_from_db()
+		self.assertEqual(pr.status, "Paid")
+		self.assertEqual(pr.outstanding_amount, 0)
+		self.assertEqual(pr.grand_total, 1000)
+
+		# creating a more payment Request must not allowed
+		self.assertRaisesRegex(
+			frappe.exceptions.ValidationError,
+			re.compile(r"Payment Request is already created"),
+			make_payment_request,
+			dt="Sales Order",
+			dn=so.name,
+			mute_email=1,
+			submit_doc=1,
+			return_doc=1,
+		)
+
+	def test_multiple_payment_if_partially_paid_for_multi_currency(self):
+		pi = make_purchase_invoice(currency="USD", conversion_rate=50, qty=1, rate=100)
```

Comment:
@Abdeali099 @vorasmit 

Is this PI supposed to post to company currency account or USD account?

Reply:
Company Currency Account.

Reply:
To give some context, I was trying to refactor the Journal Entry test suite (https://github.com/frappe/erpnext/pull/43405) to make use of `FrappeTestCase` fixture and some test cases in Payment Request started to fail https://github.com/frappe/erpnext/actions/runs/11102491305/job/30842389912?pr=43405#step:13:716

> Company Currency Account.


In that case, the test name along with passing 'USD' as currency to the make_purchase_invoice() cause a misleading understanding.